Common/src/grid_movement_structure.cpp

```diff
@@ -3895,79 +4234,104 @@ void CSurfaceMovement::SetFFDCPChange_2D(CGeometry *geometry, CConfig *config, C
       FFDBox->SetControlPoints(index, movement);
     }
   }
-		
+  else {
+    return false;
+  }
+  
+  return true;
+  
 }
 
-void CSurfaceMovement::SetFFDCPChange(CGeometry *geometry, CConfig *config, CFreeFormDefBox *FFDBox,
-																			unsigned short iDV, bool ResetDef) {
-	
-	su2double movement[3] = {0.0,0.0,0.0}, Ampl;
-	unsigned short index[3], i, j, k, iPlane;
-	string design_FFDBox;
-
+bool CSurfaceMovement::SetFFDCPChange(CGeometry *geometry, CConfig *config, CFreeFormDefBox *FFDBox, CFreeFormDefBox **ResetFFDBox,
+                                      unsigned short iDV, bool ResetDef) {
+  
+  su2double movement[3] = {0.0,0.0,0.0}, Ampl;
+  unsigned short index[3], i, j, k, iPlane, iFFDBox;
+  bool CheckIndex;
+  string design_FFDBox;
+  su2double Scale = config->GetFFD_Scale();
```

Comment:
Long story short. This scale factor is a method to relate the value of the design variable with a meaningful movement of the control points (in a design process). It is more or less equivalent to the scale factor that we apply in OPT_OBJECTIVE or OPT_CONSTRAINT.